docs/user-guide/advanced_topics/dynamic_federations.md

```diff
@@ -0,0 +1,46 @@
+
+Apart from time considerations there are no additional requirements. The checks on the subscription are done immediately as opposed to waiting for the `enterInitializationMode` call.
+
+### Dynamic Observer Federates
+
+A federate may declare itself to be and observer in the FederateInfo Structure when a federate is declared. This can be done via command line (`--observer`) or through a flag.
+
+In the C++ API
+
+```c++
+FederateInfo fi;
+fi.observer=true;
+```
+
+and in the C or other language API's
+
+```c
+helicsFederateInfoSetFlagOption(fi, HELICS_FLAG_OBSERVER,HELICS_TRUE,&err);
+```
+
+The observer flag triggers some flags in the created core and federate to notify the broker that it can be dynamically added. Otherwise it will get an error notification that the federate is not accepting new federates. If a core is created before the new federate it must also be created with the observer flag enabled. Otherwise, it will not be allowed to join the federation.
+
+Once joined, subscriptions can be added with the same timing rules as described in the previous section. One key thing to be aware of is that for dynamic federates the time returned after enterExecutingMode is not necessarily 0, but will depend on the time of federates containing the publications or endpoints that are being linked. If there are no data pathways zero will still be returned.
+
+The utility of this capability is primarily for debugging/observation purposes. It is possible to join, get the latest data and make some queries about the current co-simulation status for monitoring purposes and disconnect. At present a new name is required each time an observer connects. This may be relaxed in the future.
+
+### Dynamic Publications
+
+Dynamic publications and endpoints will be allowed in the near future. These would include just creating a publication or endpoint on an already executing federate. It would then be available for others to connect to and receive data. It would typically take at least two timesteps of the publisher to actually receive any data from the new publication. The first to ensure registration is completed and the second publish the data. It would be possible to send the data immediately after the register call but it would be impossible(unless on the same core) for any other federate to connect until after the second request time call.
```

Comment:
```suggestion
Dynamic publications and endpoints will be allowed in the near future. These would include just creating a publication or endpoint on an already executing federate. It would then be available for others to connect to and receive data. It would typically take at least two timesteps of the publisher to actually receive any data from the new publication. The first to ensure registration is completed and the second to publish the data. It would be possible to send the data immediately after the register call but it would be impossible (unless on the same core) for any other federate to connect until after the second request time call.
```